app/actions/websocket.ts

```diff
@@ -958,6 +962,21 @@ function handleChannelSchemeUpdatedEvent(msg: WebSocketMessage) {
     };
 }
 
+function handleUpdateMemberRoleEvent(msg: WebSocketMessage) {
+    return async (dispatch: DispatchFunc) => {
+        const memberData = JSON.parse(msg.data.member);
+        const newRoles = memberData.roles.split(' ');
+
+        dispatch(loadRolesIfNeeded(newRoles));
+
+        dispatch({
```

Comment:
once you apply the changes by removing `loadRolesIfNeeded` you could batch the actions.

Reply:
Done.